psc-ide/PROTOCOL.md

```diff
@@ -560,6 +560,43 @@ Valid namespaces are `value`, `type` and `kind`.
 }
 ```
 
+### Declaration type filter
+A filter which allows to filter type declarations. Valid type declarations are
+`value`, `type`, `synonym`, `constructor`, `typeclass`, `valueoperator`,
+`typeoperator` and `kind`.
+
+```json
+{
+  "filter": "declarations",
+  "params": [
+    {
+      "declarationtype": "value"
+    },
+    {
+      "declarationtype": "type"
+    },
+    {
+      "declarationtype": "synonym"
+    },
+    {
+      "declarationtype": "constructor"
```

Comment:
I'd prefer dataconstructor here, since we also have type constructors, which could make this slighty confusing.